clang/include/clang/Basic/Attr.td

```diff
@@ -1622,6 +1636,33 @@ def SYCLAddIRAttrCommonMembers : SYCLAddIRAttrMemberCodeHolder<[{
                                       ValueE->getType()->isSignedIntegerType());
       return std::string(IntegerStrBuffer);
     }
+    if (const auto *InitListE = dyn_cast<InitListExpr>(ValueE)) {
+      if (InitListE->isStringLiteralInit()) {
+        const Expr *StringInitE = InitListE->getInit(0)->IgnoreParenImpCasts();
+        return getValidAttributeValueAsString(StringInitE, Context);
+      }
+
+      SmallString<10> StrBuffer;
+      for (const auto *InitE : InitListE->inits()) {
+        const Expr *InitNoImpCastE = InitE->IgnoreParenImpCasts();
+        char C = 0;
+        if (const auto *CharacterVal =
+              dyn_cast<CharacterLiteral>(InitNoImpCastE))
+          C = static_cast<char>(CharacterVal->getValue());
+        else if (const auto *IntegerVal =
+              dyn_cast<IntegerLiteral>(InitNoImpCastE))
+          C = static_cast<char>(IntegerVal->getValue().getZExtValue());
+        else
```

Comment:
So, only character and integer are supported? Can part of property value be a `double`?

Reply:
The design doc for compile-time properties allows floating point property values also, but I don't think we currently have any like that.  If this PR is intending to implement the general support for compile-time properties, we should support FP types also:

https://github.com/intel/llvm/blob/sycl/sycl/doc/design/CompileTimeProperties.md#property-representation-in-c-attributes-and-in-ir

Reply:
I think there's a bit of confusion here. This is only for parsing a string of characters defined as an initializer list. For this we should only need to be able to parse actual characters and integral values. Float, bool, etc are handled separately.

Reply:
I see. But how we can do "support" (as you mentioned here https://github.com/intel/llvm/pull/6941#discussion_r987246940 ) of multi value property if we would like it to have a non-char/int as one of the values?